cloud: add TiDB Cloud CLI docs (#12111)
pingcap/docs: tidb-cloud/ticloud-cluster-create.md

---
title: ticloud cluster create
Summary: The reference of `ticloud cluster create`.
---

# ticloud cluster create

Create one cluster in the specified project

```shell
ticloud cluster create [flags]
```

## Examples

Create a cluster in interactive mode:

```shell
ticloud cluster create
```

Create a cluster in non-interactive mode:

```shell
ticloud cluster create --project-id <project-id> --cluster-name <cluster-name> --cloud-provider <cloud-provider> --region <region> --root-password <password> --cluster-type <cluster-type>
```

## Flags

| Flag                    | Description                                                 | Required | Extra                             |
|-------------------------|-------------------------------------------------------------|----------|-----------------------------------|

> | --cloud-provider string | Cloud provider, one of [AWS]                                | Yes      | Only work in non-interactive mode |

Reviewer: Does "one of [AWS]" mean that TiDB Cloud CLI only supports creating clusters in AWS,  not in Google Cloud?

Reviewer 2: Currently users could only use CLI to create **serverless clusters**. Serverless clusters only support AWS.

> | --cluster-name string   | Name of the cluster to de created                           | Yes      | Only work in non-interactive mode | 

Reviewer: ```suggestion
| --cluster-name string   | Name of the cluster to be created                           | Yes      | Only work in non-interactive mode | 
```

> | --cluster-type string   | Cluster type, only support "SERVERLESS" now                 | Yes      | Only work in non-interactive mode | 

Reviewer: ```suggestion
| --cluster-type string   | Cluster type. Currently, only "SERVERLESS" is supported.     | Yes      | Only work in non-interactive mode | 
```

> | -h, --help              | help for create                                             |          |                                   |

Reviewer: ```suggestion
| -h, --help              | Get the help information                |   No  |     Work in both non-interactive and interactive modes.     |                                   |
```

| -p, --project-id string | The ID of the project, in which the cluster will be created | Yes      | Only work in non-interactive mode | 
| -r, --region string     | Cloud region                                                | Yes      | Only work in non-interactive mode | 
| --root-password string  | The root password of the cluster                            | Yes      | Only work in non-interactive mode | 

> <Note> For flags required in non-interactive mode, fill them according to the prompt in interactive mode. </Note>

Reviewer: ```suggestion
For flags required in interactive mode, fill them according to the prompt in interactive mode.
```

Reviewer 2: If change this, don't forget to change all other places.
